vendor/bat-native-ads/src/bat/ads/ad_content_info.cc

```diff
@@ -53,240 +53,105 @@ AdContentLikeActionType AdContentInfo::ToggleThumbDownActionType() const {
-      dictionary->FindStringKey("campaignId");
-  if (campaign_id_value) {
-    campaign_id = *campaign_id_value;
+  if (const auto* value = root.FindString("campaign_id")) {
+    campaign_id = *value;
   }
 
-  const std::string* advertiser_id_value =
-      dictionary->FindStringKey("advertiserId");
-  if (advertiser_id_value) {
-    advertiser_id = *advertiser_id_value;
+  if (const auto* value = root.FindString("advertiser_id")) {
+    advertiser_id = *value;
   }
 
-  const std::string* brand_value = dictionary->FindStringKey("brand");
-  if (brand_value) {
-    brand = *brand_value;
+  if (const auto* value = root.FindString("brand")) {
+    brand = *value;
   }
 
-  const std::string* brand_info_value = dictionary->FindStringKey("brandInfo");
-  if (brand_info_value) {
-    brand_info = *brand_info_value;
+  if (const auto* value = root.FindString("brand_info")) {
+    brand_info = *value;
   }
 
-  const std::string* brand_display_url_value =
-      dictionary->FindStringKey("brandDisplayUrl");
-  if (brand_display_url_value) {
-    brand_display_url = *brand_display_url_value;
+  if (const auto* value = root.FindString("brand_display_url")) {
+    brand_display_url = *value;
   }
 
-  const std::string* brand_url_value = dictionary->FindStringKey("brandUrl");
-  if (brand_url_value) {
-    brand_url = GURL(*brand_url_value);
+  if (const auto* value = root.FindString("brand_url")) {
+    brand_url = GURL(*value);
   }
 
-  const absl::optional<int> like_action_type_optional =
-      dictionary->FindIntKey("likeAction");
-  if (like_action_type_optional) {
-    like_action_type =
-        static_cast<AdContentLikeActionType>(like_action_type_optional.value());
+  if (const auto value = root.FindInt("like_action")) {
+    like_action_type = static_cast<AdContentLikeActionType>(*value);
   }
 
-  const std::string* confirmation_type_value =
-      dictionary->FindStringKey("adAction");
-  if (confirmation_type_value) {
-    confirmation_type = ConfirmationType(*confirmation_type_value);
+  if (const auto* value = root.FindString("ad_action")) {
+    confirmation_type = ConfirmationType(*value);
   }
 
-  const absl::optional<bool> is_saved_optional =
-      dictionary->FindBoolKey("savedAd");
-  if (is_saved_optional) {
-    is_saved = is_saved_optional.value();
+  if (const auto value = root.FindBool("saved_ad")) {
+    is_saved = *value;
   }
 
-  const absl::optional<bool> is_flagged_optional =
-      dictionary->FindBoolKey("flaggedAd");
-  if (is_flagged_optional) {
-    is_flagged = is_flagged_optional.value();
+  if (const auto value = root.FindBool("flagged_ad")) {
+    is_flagged = *value;
   }
 
   return true;
 }
 
 std::string AdContentInfo::ToJson() const {
   std::string json;
-  SaveToJson(*this, &json);
+  CHECK(base::JSONWriter::Write(ToValue(), &json));
   return json;
 }
 
 bool AdContentInfo::FromJson(const std::string& json) {
-  rapidjson::Document document;
-  document.Parse(json.c_str());
+  auto document =
```

Comment:
can we use `absl::optional<base::Value>` instead of auto for ads changes.

Reply:
Fixed in 32eb3390638af962d9af9b6b989e2488f2151c10.